eeschema/bus-wire-junction.cpp

```diff
@@ -650,33 +730,39 @@ bool SCH_EDIT_FRAME::SchematicCleanUp( bool aAppend )
         }
     }
 
-    for( item = screen->GetDrawItems(); item; item = secondItem )
+    for( item = aScreen->GetDrawItems(); item; item = secondItem )
     {
         secondItem = item->Next();
 
         if( item->GetFlags() & STRUCT_DELETED )
-            RemoveFromScreen( item );
+            RemoveFromScreen( item, aScreen );
     }
 
-    SaveCopyInUndoList( itemList, UR_CHANGED, aAppend );
+    if( itemList.GetCount() && aUndo )
+        SaveCopyInUndoList( itemList, UR_DELETED, true );
```

Comment:
Is this line correct?  You can AddToScreen() above, so I thought UR_CHANGED made sense.

Reply:
The `UNDO_REDO_T` argument is basically ignored in `SaveCopyInUndoList(PICKED_ITEMS_LIST&)` since each item in the list will also have a paired command.  It is only used when an item in the list has change type `UR_UNSPECIFIED`.  I changed it away from `UR_CHANGED` because that seemed even more wrong than deleted or added, since `UR_CHANGED` is used for when parameter values of an object are changed, which isn't happening here under any circumstance.

Reply:
Got it.  I think I picked UR_CHANGED because it makes a copy of the element as it was before (the other actions don't) but you are right, we shouldn't need this.